src/prefect/server/schemas/responses.py

```diff
@@ -261,18 +265,36 @@ class DeploymentResponse(ORMBaseModel):
         default=None,
         description="The name of the deployment's work pool.",
     )
+    status: Optional[schemas.statuses.DeploymentStatus] = Field(
+        default=schemas.statuses.DeploymentStatus.NOT_READY,
+        description="Whether the deployment is ready to run flows.",
+    )
     enforce_parameter_schema: bool = FieldFrom(schemas.core.Deployment)
 
     @classmethod
     def from_orm(
         cls, orm_deployment: "prefect.server.database.orm_models.ORMDeployment"
     ):
         response = super().from_orm(orm_deployment)
+
         if orm_deployment.work_queue:
             response.work_queue_name = orm_deployment.work_queue.name
             if orm_deployment.work_queue.work_pool:
                 response.work_pool_name = orm_deployment.work_queue.work_pool.name
 
+        offline_horizon = datetime.datetime.now(
+            tz=datetime.timezone.utc
+        ) - datetime.timedelta(seconds=DEPLOYMENT_LAST_POLLED_TIMEOUT_SECONDS)
+
+        if response.last_polled and response.last_polled > offline_horizon:
+            response.status = schemas.statuses.DeploymentStatus.READY
+        elif (
+            orm_deployment.work_queue
+            and orm_deployment.work_queue.last_polled
+            and orm_deployment.work_queue.last_polled > offline_horizon
+        ):
+            response.status = schemas.statuses.DeploymentStatus.READY
```

Comment:
I felt it was more readable to separate these conditions but welcome other thoughts